pkg/kubelet/config/config.go

```diff
@@ -235,6 +240,10 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 				ref.Annotations = make(map[string]string)
 			}
 			ref.Annotations[kubetypes.ConfigSourceAnnotationKey] = source
+			// ignore static pods
+			if ref.ResourceVersion != "" {
```

Comment:
How about changing this to explicit if for "source", e.g. filtering for source set to value specific to apiserver?

Reply:
Good point. I changed it to use kubetype helper.